[MRG] CalibratedClassifierCV passes on fit_params

BenjaminBossan opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Partly addresses #12384

#### What does this implement/fix? Explain your changes.

This PR makes it possible to pass `fit_params` to the fit method of the
`CalibratedClassifierCV`, which are then routed to the underlying base
estimator.

*Note*: I implemented `predict_proba` on `CheckingClassifier` for the
new unit test to work.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

While working on this issue, I discovered these lines, which may be problematic:

https://github.com/scikit-learn/scikit-learn/blob/86aea99159c944fae1f519ac61c8b8c484449130/sklearn/calibration.py#L168-L176

In fact, if the base estimator doesn't have `sample_weight` in
its signature, the `sample_weight` argument is not routed to
it. However, one could argue that it should still be routed to it
if `fit_params` are part of the signature.

This can be relevant, for instance, when the base estimator is
just a meta estimator that delegates all `fit_params` to the true
estimator. With the current implementation, `sample_weight` would
not be passed on.

Commit:
CalibratedClassifierCV passes on fit_params

Partly addresses to https://github.com/scikit-learn/scikit-learn/issues/12325

This PR makes it possible to pass fit_params to the fit method of the
CalibratedClassifierCV, which are then routed to the underlying base
estimator.

Commit:
Add test for CheckingClassifier

After adding the predict_proba method on CheckingClassifier, the test
coverage decreased. Therefore, a test for predict_proba was
added. More tests for CheckingClassifier will be added in a separate
PR.

BenjaminBossan commented:
The test coverage decreased after adding `predict_proba` in `CheckingClassifier`, which is why I added a test for it. I will add more tests for `CheckingClassifier` in a separate PR.

Commit:
Remove unnecessary test code

Commit:
Rename function used in test

ogrisel commented on sklearn/utils/_mocking.py:
```diff
@@ -115,6 +115,20 @@ def predict(self, T):
             assert self.check_X(T)
         return self.classes_[np.zeros(_num_samples(T), dtype=np.int)]
 
+    def predict_proba(self, T):
```
why `T`? The scikit-learn convention is `X` for the input data of a statistical estimator.

BenjaminBossan replied:
I tried to stick close to the `predict` method, which also uses `T`. But I can change it (as well as `predict`).

rth replied:
please do change it.

ogrisel reviewed (approved):
LGTM, just a quick nitpick:

jnothman commented on sklearn/tests/test_calibration.py:
```diff
@@ -341,3 +342,21 @@ def decision_function(self, X):
+    X, y = make_classification(n_samples=2 * n_samples, n_features=6,
+                               random_state=42)
+
+    fit_params = {'a': y, 'b': y}
```
Please also check with non-arrays, such as lists

BenjaminBossan replied:
What would be the expectation here? That lists work as long as they have the correct length? Is there something else that should work?

jnothman replied:
Yes, lists are the main thing to check, under the assumption that if they are handled, so are other sequences

BenjaminBossan replied:
With lists, the line `fit_params_train[k] = v[train]` fails, as expected. I thus wanted to use `indexable` to make sure that lists would pass but actually, `indexable` doesn't touch lists because of this line:

https://github.com/scikit-learn/scikit-learn/blob/1495f69242646d239d89a5713982946b8ffcf9d9/sklearn/utils/validation.py#L224-L225

What is the right way to deal with lists in this case?

jnothman replied:
So I think you need to use safe_indexing rather than `v[train]`

BenjaminBossan replied:
Thx, I forgot about that function. I used `_safe_indexing` instead of `safe_indexing` because the latter is deprecated.

hs-nazuna committed:
DOC add explanation of n_jobs in sklearn/model_selection/_validation.py (#17717)

thomasjpfan committed:
DOC Removes FunctionTransformer example (#17318)

* DOC Removes FunctionTransformer example

* DOC Updates description

Commit:
DOC Add explanation of n_jobs in gaussian_process/_gpc.py (#17716)

thomasjpfan committed:
DOC Adds note on dataframe in contributing guide (#17359)

* DOC Adds note on dataframe

* DOC Address comments

thomasjpfan committed:
ENH Patches sphinx.ext.autosummary for case insensitive file systems (#13022)

* ENH: Patches autosummary for case insensitive file systems

* DOC: More details

* REV

adrinjalali committed:
Revert "ENH Uses binned values from training to find missing values (#16883)"

This reverts commit e5cc2b0e20df35a890c125fdc057668330450ebf.

postmalloc committed:
ENH Add inverse_transform feature to SimpleImputer (#17612)

hs-nazuna committed:
DOC add explanation of n_jobs in permutation_importance (#17723)

thomasjpfan committed:
MNT Fix extension for sphinx 3.0 (#17724)

beatrizsmg committed:
DOC improve attributes and parameters descriptions in MultiLabelBinarizer (#17718)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

rth committed:
MNT Add type annotations for OpenML fetcher (#17053)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

kcetina-fujitsu committed:
DOC improve attributes and parameters descriptions in RFE and RFECV (#17721)

Co-authored-by: Beatriz San Miguel <beatriz.sanmiguelgonzalez@uk.fujitsu.com>
Co-authored-by: Loïc Estève <loic.esteve@ymail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

rubywerman committed:
DOC improve description of OneClassSVM when used for outlier detection (#17726)

marenwestermann committed:
DOC coef_ and intercept_ documentation for ComplementNB (#17722)

kishimoto-banana committed:
[MRG] Expose interpolation thresholds as public fitted attribute of IsotonicRegression (#16289)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Thomas J Fan <thomasjpfan@gmail.com>
Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

sdpython committed:
BUG fix SparseCoder to follow scikit-learn API and allow cloning (#17679)

* BUG fix SparseCoder to follow scikit-learn API

* TST check that get_params and set_params work as expected

* address comments

* PEP8

* iter

* Fixes #8675, fix cloning for SparseCoder

* remove spaces

* Update _dict_learning.py

* fix confusing arguments

* remove unnecessary code

* Update test_common.py

* removes spaces

* PEP8

* iter

* fix merge

* ignore a mypy warning

* type: ignore

* remove one deprecated verification

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

* Update sklearn/decomposition/tests/test_dict_learning.py

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

* test clone produces different id

* review

* add one more test

* lint

* Update sklearn/decomposition/tests/test_dict_learning.py

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

shashanksingh28 committed:
ENH Verify md5-checksums received from openml arff file metadata(#14800)

Co-authored-by: Thomas J Fan <thomasjpfan@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

j2heng committed:
DOC add missing attributes to LocalOutlierFactor (#17696)

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lucyleeow committed:
DOC Add example in docstring calibration_curve (#17731)

glemaitre committed:
DOC add whats new entry following #17679 (#17738)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

thomasjpfan committed:
ENH _fit_and_score now returns a dictionary (#17332)

* ENH _fit_and_score now returns a dictionary

* MRG

* REV

* LOL

* Removing things in gitignore is fun

alfaro96 committed:
MNT More replacements of numpy aliases with built-in types (#17707)

* MNT More replacements of numpy aliases with built-in types [scipy-dev]

* MNT More (manual) replacements of numpy aliases

* MNT More (manual) replacements of numpy aliases

* FIX Minor change

* MNT Trigger build [scipy-dev]

* FIX Minor change

* MNT Trigger build [scipy-dev]

* FIX More deprecations of numpy aliases [scipy-dev]

* MNT Silent numpy aliases warnings [scipy-dev]

* FIX Fix silent deprecations

* Trigger build [scipy-dev]

* FIX Add scape characters [scipy-dev]

* FIX Remove package specification from silent

* Trigger build [scipy-dev]

* Revert

maikia committed:
TST check equivalence normalize/StandardScaler and dense/sparse in linear models (#17665)

Co-authored-by: Alexandre Gramfort <alexandre.gramfort@m4x.org>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

kcetina-fujitsu committed:
DOC improve attributes and parameters descriptions in KernelPCA (#17747)

Co-authored-by: Beatriz San Miguel <beatriz.sanmiguelgonzalez@uk.fujitsu.com>
Co-authored-by: Loïc Estève <loic.esteve@ymail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

Commit:
FIX fix multinomial deviance by taking the weighted average instead of the sum (#17694)

Co-authored-by: Markus Rempfler <markus.rempfler@tum.de>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

rubywerman committed:
DOC Change remove symbol nu from OutlierDetection (#17732)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

alfaro96 committed:
TST Skip test_common_check_return_X_y if pillow not installed (#17692)

avigupta2612 committed:
ENH Add parameter in colorbar in plot_confusion_matrix (#17192)

Co-authored-by: Venkatachalam N <venky.yuvy@gmail.com>

jeremiedbb committed:
BUILD Make sure PKG-INFO contains Requires-Python (#17683)

alfaro96 committed:
FIX Extract estimator objects before aggregating dict of scores (#17745)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

lschwetlick committed:
ENH Enable percentage for n_features_to_select in RFE(Fix PR #16228) (#17090)

* added code from PR #14627

* fixed error handling None as n_features_to_select

* added test for error message and percentage passing

* linting

* more linting

* even more linting

* sklearn/feature_selection/_rfe.py make int casting simpler

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* exchange redundant test case with small float

* add what's new

* removed useless import

* lint

* Fix whats_new/v0.24.rst

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* move whats new section

* disambiguate choice between 1 feature and 100% of features

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* move checking for negatives to _fit

* update error to include nones

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* markdown style in what's new

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* Update sklearn/feature_selection/tests/test_rfe.py

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* forgot to delete line

* fix test for negative n_features error

* BUG Fixes issues

* added case for float >1 and more detailed error messages

* lint

* more lint

* more lint

* CLN Minor adjustments

* BLD Force build on ci

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
TST Fix estimator checks for classifers with poor_score tag (#16851)

postmalloc committed:
MNT Remove unused variable `batch_size` in MLP `_fit` (#17754)

thomasjpfan committed:
MNT Places all minimum dependencies in one central location (#17573)

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

justmarkham committed:
DOC Update links on Contributing page (#17757)

cmarmo committed:
DOC Update what's new for version 0.24 (#17755)

Justin-Huber committed:
TST Python wrapper and tests for sklearn/svm/newrand/ (#17157)

* Implemented issue #16862

* Implemented issue #16862

* Ready for pull request

* Cleaned up redundancies

* Removed test_set_seed_wrap for now

* Update sklearn/svm/tests/test_bounds.py

Co-authored-by: smarie <sylvain.marie@schneider-electric.com>

* Review feedback changes

* Style fixes

* np.quantile -> np.percentile

* Update sklearn/svm/src/newrand/newrand.h

Merged suggestion

Co-authored-by: smarie <sylvain.marie@schneider-electric.com>

* Update sklearn/svm/tests/test_bounds.py

Merge suggestions

Co-authored-by: smarie <sylvain.marie@schneider-electric.com>

* linting fix

* linting fix

* linting fix

* Added deterministic test for mt19937 in svm/newrand.h

* Linting fix

* Review changes

* Update sklearn/svm/tests/test_bounds.py

Co-authored-by: smarie <sylvain.marie@schneider-electric.com>

* Update sklearn/svm/tests/test_bounds.py

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

* Update sklearn/svm/tests/test_bounds.py

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

* Update test_bounds.py

* Linting fixes

Co-authored-by: smarie <sylvain.marie@schneider-electric.com>
Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

alfaro96 committed:
MNT Use central location of dependencies for travis-ci (#17761)

thomasjpfan committed:
DOC Fix some glossary links (#17773)

chkoar committed:
TST Refactor the test of oob score of forest predictors (#17770)

chkoar committed:
CLN Remove self-assignments from tests (#17776)

alfaro96 committed:
FIX Specify C++11 compiler for sklearn/svm/_newrand (#17767)

thomasjpfan committed:
CI Correctly get commit message in azure pipeline (#17688)

* CI Correctly gets commit message for pull request

* BLD [scipy-dev]

* BLD [scipy-dev]

* CI Debug

* CI Debug [scipy-dev]

* CI Debug [scipy-dev]

* CI Debug another [scipy-dev]

* CI Debug another2 [scipy-dev]

* CI Debug another3 [scipy-dev]

* CI Debug another4 [scipy-dev]

* CI Debug another5 [scipy-dev]

* CI Debug another6 [scipy-dev]

* CI Debug 7 [scipy-dev]

* CI Debug 8 [scipy-dev]

* CI Debug 9 [scipy-dev]

* CI Debug 10 [scipy-dev]

* CI Debug 11 [scipy-dev]

* CI Debug 12 [scipy-dev]

* CI Debug 13 [scipy-dev]

* CI Debug 14 [scipy-dev]

* CI Debug 15 [scipy-dev]

* CI Debug 16 [scipy-dev]

* CI Debug 17 [scipy-dev]

* CI Debug 18 [scipy-dev]

* CI Debug fix [scipy-dev]

* CI Debug 19 [scipy-dev]

* CI Debug 20 [scipy-dev]

* REV Revert [scipy-dev]

* CI Debug 21 [scipy-dev] lint skip

* CI Debug 22 [scipy-dev]

* CI Debug 22 [scipy-dev] [lint skip]

* CI Debug 23 [scipy-dev] [lint skip]

* CI Debug 24 [scipy-dev] [lint skip]

* CI Debug 25 [scipy-dev] [lint skip]

* CI Debug 26 [scipy-dev] [lint skip]

* CI Debug 28 [scipy-dev] [lint skip]

* CI Debug 29 [scipy-dev] [lint skip]

* CI Debug 30 [scipy-dev] [lint skip]

* CI Debug 31 [scipy-dev] [lint skip]

* CI Debug 32 [scipy-dev] [lint skip]

* CI Debug 33 [scipy-dev] [lint skip]

* BLD [scipy-dev]

lucyleeow committed:
ENH Support pipelines in CalibratedClassifierCV (#17546)

alexhenrie committed:
PERF Discard old layers immediately when making neural net predictions (#17661)

thomasjpfan committed:
Revert "ENH Patches sphinx.ext.autosummary for case insensitive file systems (#13022)" (#17780)

This reverts commit 8a695d7d857ebc63b91ccef8315b19c2528494e5.

NicolasHug committed:
Fix SVs computation (#17405)

chkoar committed:
MNT Collapse an unnecessary enclosed if statement (#17766)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

postmalloc committed:
ENH 32-bit support for MLPClassifier and MLPRegressor (#17759)

postmalloc committed:
PERF Dot product optimizations using multi_dot (#17737)

marenwestermann committed:
DOC coef_ and intercept_ doc for BernoulliNB (#17823)

barankarakus committed:
Fixed typo in docstring of enet_path in _coordinate_descent.py (#17807)

jeremiedbb committed:
TST Fix test_float_precision fails on arm (#17802)

ashutosh1919 committed:
Added mean_absolute_percentage_error in metrics fixes #10708 (#15007)

Co-authored-by: mohamed-ali <m.ali.jamaoui@gmail.com>
Co-authored-by: Alexandre Gramfort <alexandre.gramfort@m4x.org>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Joel Nothman <joel.nothman@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@pm.me>

alfaro96 committed:
MNT Prevent coverage status update on each upload to codecov (#17822)

Thomas9292 committed:
DOC Add tiebreaking explanation to DecisionTreeClassifier docs (#17783)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>
Co-authored-by: thomas.wesselink@ing.com <thomas.wesselink@ing.com>

cool-RR committed:
MNT use raise from in validation.py (#17545)

alfaro96 committed:
MNT Deprecate coef_ and intercept_ in naive bayes estimators (#17427)

alfaro96 committed:
FIX Change MRO for some estimators (#17837)

jona-sassenhagen committed:
[MRG] varimax rotation for factor analysis (#11064)

* init varimax

* fix py2 and add example text

* fix print again

* test, docs

* fix test

* fix doctstring

* np.allclose for test

* add refrence

* address comments

* update

* Fix linting failure.

* Fix deprecation warning.

* a bit of cleanup

* Address @jnothman's review comments.

* Update section divider

* Add note in what's new page and versionadded to docstring.

Co-authored-by: Eric Prestat <eric.prestat@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

adrinjalali committed:
MNT deprecate ChangedBehaviorWarning and NonBLASDotWarning (#17804)

* deprecate ChangedBehaviorWarning and NonBLASDotWarning

* add whats_new entry

acharles7 committed:
MNT Removed arguments from matplotlib.use() (#17774)

fujiaxiang committed:
ENH Allow isotonic reg 2d input with 1 feature (GH15012) (#17379)

Commit:
DOC Add missing attrributes in ElasticNet and etc (#17740)

Co-authored-by: BenjaminLiu <imbenliu@gmail.com>
Co-authored-by: Loïc Estève <loic.esteve@ymail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

claramatos committed:
EHN Provide a pos_label parameter in plot_roc_curve (#17651)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Commit:
DOC Updated cross validation accuracy metric (#17781)

* Updated cross validation accuracy metric

* Added deterministic results

* Update std value

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

* Revert output

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

brcharron committed:
FIX Fix handling of binary_only tag in check_estimator (#17812)

Co-authored-by: Bruno Charron <bruno@charron.email>

jeremiedbb committed:
FIX Do not normalize sample weights in KMeans (#17848)

`sample_weight` should not be normalized in KMeans. The weight magnitude
should have an influence on the `inertia_`, larger the weights, larger should be
the inertia.

jeremiedbb committed:
TST clean up and parametrization in KMeans tests (#17850)

glemaitre committed:
TST remove useless network marker in GBDT tree (#17858)

gimseng committed:
FIX Raise error when l1_ratio is out-of-range in ElasticNet (#17846)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

glemaitre committed:
TST parametrize and clean-up tests GBDT (#17860)

adrinjalali committed:
DOC use conda-forge in installation guideline (#17825)

Ultramann committed:
MNT Factor out concatenation of Xs after feature union transformations (#17868)

thomasjpfan committed:
MNT Small refactor for cross_validate (#17867)

thomasjpfan committed:
DEP deprecate grid_score_ and cv_alphas_ in GraphicalLassoCV (#16392)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

thomasjpfan committed:
BUG Fixes stacker compatibility with estimators without n_features_in_ (#17357)

thomasjpfan committed:
CLN Minor adjustment for potential overflow in hist gradient boosting (#17410)

NicolasHug committed:
DOC minor update to classification_report docstring (#16899)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Andreas Mueller <t3kcit@gmail.com>

thomasjpfan committed:
ENH Makes _deprecate_positional_args slightly faster (#17246)

alfaro96 committed:
DOC Specify shape with parentheses in decision tree guide (#17873)

thomasjpfan committed:
CLN Uses bool instead of np.bool (#17875)

jeremiedbb committed:
MNT Centralize all param checks in KMeans (#17861)

strivn committed:
DOC Fix documentation of default values in tree exports (#17834)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

yashika51 committed:
DOC Added missing attributes to OneVsRest Classifier (#17844)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Fix default values in multioutput.py docstrings (#17841)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
FIX MeanShift when estimated bandwidth is 0 (#17742)

shail2512-lm10 committed:
DOC added attributes in LassoLarsCV (#17838)

glemaitre committed:
EXA Improve the alphas values in MLP example (#17880)

Co-authored-by: 叶贞明 TaiChiTiger <sakyayj@126.com>

raybellwaves committed:
DOC add Gitter link to readme (#17883)

jeremiedbb committed:
CI Fix pre-build checks to handle compilers from build_ext options (#16193)

jeremiedbb committed:
CI import setuptools monkeypatch distutils commands (#17887)

yashika51 committed:
ENH Added parameter clip to MinMaxScaler (#17833)

alfaro96 committed:
DOC Specify shape with parentheses in ensemble user guide (#17888)

marenwestermann committed:
add dual_gap_ attribute (#17882)

alfaro96 committed:
DOC Add Twitter link to README.rst (#17891)

biprateep committed:
removed os import (#17893)

biprateep committed:
MNT removed repeated definition of ranked_exposure (#17895)

willpeppo committed:
EXA Avoid redundant numpy imports in plot_release_highlights.py (#17903)

mikeaalv committed:
ENH remove default y value in sklearn.kernel_ridge.KernelRidge.fit (#17906)

biprateep committed:
MNT Fixed import statements (#17901)

* fixed import statements

* fixed linting issues

* keep numpy import within setup_module

Carreau committed:
DOC Fix numpydoc section name Return -> Returns (#17879)

From a cursory glance this is the only place I found in sklearn (and to a wider extend scipy/numpy) that uses Return instead of Returns as described in the numpydoc format.

mikeaalv committed:
[MRG] update the documents (default value) in the function sklearn.kernel_ridge.KernelRidge (#17904)

* update the documents (default value) in the function kernel_ridges.py

* update the array size notation

* update sample_weight array size

Hoda1394 committed:
DOC updated documentation default value for gradient_boosting (#17909)

joshuacwnewton committed:
[MRG] ENH Add parameter to sklearn.manifold.TSNE to transition to squaring non-Euclidean distance metrics by default (#12401) (#17662)

jeremiedbb committed:
DOC document and deprecate missing attributes in MiniBatchKMeans (#17864)

Probinette4 committed:
DOC update to text/CountVectorizer token_pattern parameter default value (#17898)

* Updated documenation in CountVectorizer function for the token_parameter default value

* Removed the r before the string, error in flake8

* Additional change to see if flake8 error solved

Co-authored-by: Preston <pkrobinet@presby.edu>

franslarsson committed:
DOC Fix documentation of default values in sklearn.utils.optimize.py (#17881)

* update docstring according to guideline

* add space for consistency

franslarsson committed:
DOC Fix documentation of default values in sklearn.utils._encode.py (#17859)

thomasjpfan committed:
CI Build with cython prerelease (#17908)

willpeppo committed:
MNT code hygiene on _mocking.py (#17894)

glemaitre committed:
FIX remove pytest warning in dataset/tests/test_common.py (#17828)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

lucyleeow committed:
DOC Clarify return of cross_val_predict (#17851)

chaitanyamogal committed:
DOC fix plot position in statistical_inference/settings.rst (#17919)

changed position of the image

thomasjpfan committed:
DOC Small doc formatting fixes in decomposition (#17918)

TahiriNadia committed:
DOC cleaning parameter docstrings in decomposition/_dict_learning.py (#17896)

* update formatting for default value for consistency'

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

* update formatting for alpha default value for consistency

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

thomasjpfan committed:
CI Temporary fix for conda-forge:: compilers on osx (#17913)

* CI Temporary fix for conda-forge on osx

* DOC Update doc on compiler version

* CLN Apply suggestions

david-cortes committed:
DOC Missing input_type to FeatureHasher (#17922)

m-sche committed:
DOC Fix error text when null_space fails with ARPACK (#17927)

alfaro96 committed:
MNT Avoid duplicate validation of data in supervised NN (#17849)

alfaro96 committed:
MNT Use substitutions for minimum dependencies in documentation (#17931)

thomasjpfan committed:
BUG Do not overwrite X when copy_X=True and Gram='auto' (#17914)

alexhenrie committed:
PERF Get rid of intermediate array in binary_log_loss (#17932)

chaitanyamogal committed:
EXA change in position of plot image (#17924)

oj-lappi committed:
DOC Fix math notation error in description of CCP (#16450)

Breiman, Friedman, Ohlsen, Stone use a T with a tilde to denote the terminal nodes of a tree. A simple T would imply we are counting the nodes in the whole tree.

Commit:
TST Add test for multinomial identifiability constraint in logistic regression (#16791)

* MNT import from numpy.testing in logistic regression

* TST add test for symmetric identifiability constraint

* address review comments

* address review comments again

* reduce atol

alfaro96 committed:
DOC Specify base_estimator=None in CalibratedClassifierCV (#17393)

thomasjpfan committed:
ENH Uses binned values from training to find missing values (#17736)

* ENH Uses training data to find missing values

* CLN Uses the binned data to find missing value

* ENH Uses binned values from training to find missing values

thomasjpfan committed:
ENH Adds support for multimetric callable return a dictionary (#15126)

marenwestermann committed:
DOC sparse_coef_ documentation for MultiTaskElasticNet and MultiTaskLasso (#17884)

jeremiedbb committed:
MNT Refactor center initialization in KMeans (#17928)

* refactor center init in KMeans

* address comments

jeremiedbb committed:
TST More cleaning/refactoring in KMeans tests (#17885)

ForrestCKoch committed:
FIX Sets dtype for pairwise_distances when metric='seuclidean' (#15730)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

SylvainLan committed:
FIX Add warning for GP's hyperparameters (#12673)

* Add a warning for GP Regressor if optimized hyperparameter matches one of its bounds

* pep8 fix

* Add a test and use np.isclose

* More verbose warning message

* remove print

* Ad a more precise warning and update tests

* fix warning and update tests

* fix test dims

* fix multiple dimensions test

* fix pep8

* change variable name

* add warning for GP classifier

* tests for gpc

* pep8 fix

* put the warning in a function

* add comment to warning function

* update name in gpr and gpc

* update whats_new

* update whats_new

* update whats_new

* remove merge conflict text

* add WhiteKernel

* put the log in the last version

alfaro96 committed:
FIX Change MRO for some estimators (#17934)

noelano committed:
PERF Call KDTree in mutual_info to reduce memory footprint (#17878)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
CI Remove unneeded check for python 3.5 (#17948)

marenwestermann committed:
FIX inconsistent hyperparameter and bounds issue in GP kernels (#17946)

thomasjpfan committed:
TST Test for both solvers in test_nmf_sparse_input (#17947)

alfaro96 committed:
DOC Fix typo in the KNeighborsClassifier documentation (#17958)

alfaro96 committed:
MNT Remove unnecessary try and except for numpy 1.3 (#17957)

jeremiedbb committed:
MNT import setuptools before distutils + add some comments (#17956)

marenwestermann committed:
DOC add attribute active_ to LarsCV (#17886)

thomasjpfan committed:
TST Moving testing code in quad_tree to test_quad_tree (#17968)

lucyleeow committed:
DOC Expand on sigmoid and isotonic in calibration.rst (#17725)

rubywerman committed:
DOC Change "Other Versions" to "Download Docs" in dropdown (#17792)

franslarsson committed:
DOC Fix documentation of defaults in sklearn.utils.class_weight (#17981)

bmaisonn committed:
DOC Fix documentation of default values in multiclass.py (#17986)

bmaisonn committed:
cleanup optional keywords in the documentation (#17990)

bmaisonn committed:
DOC Fix documentation of default values in validation.py (#17988)

amy12xx committed:
TST Added test for Incremental pca value error fix (#17985)


Co-authored-by: Allan <allanbutler9@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

saket404 committed:
DOC fix dataset loading documentation (#17993)

amy12xx committed:
FIX AffinityPropagation fix results with float32 input (#17995)

FrancescoCasalegno committed:
Distances for agglomerativeclustering (#17984)

Co-authored-by: Michael Riedmann <michael_riedmann@live.com>
Co-authored-by: Emilie Delattre <emilie.delattre@epfl.ch>
Co-authored-by: EmilieDel <47669575+EmilieDel@users.noreply.github.com>

RobRomijnders committed:
Fix typo in docstring for GaussianMixture. (#17998)

jjerphan committed:
[MRG] Use joblib.Parallel for CalibratedClassifierCV.fit (#17107)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

guillaume-chevalier committed:
DOC Add Neuraxle to Related Projects page (#16100)

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

thomasjpfan committed:
TST Default skip network tests (#18002)

mani2106 committed:
MNT Remove redundant typing import (#18004)

thomasjpfan committed:
FIX Fixes CountVectorizer sample invariance with max_features (#18016)

jeremiedbb committed:
[MRG] Fix KMeans convergence when tol==0 (#17959)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.multiclass.py (#18021)

napsternxg committed:
Fix #16998 classification_report with output_dict=True should not raise AttributeError  (#17777)

thomasjpfan committed:
CI Fixes circleci doc building by adding path to conda in bash script (#18022)

jeremiedbb committed:
MNT Add asv benchmark suite (#17026)

thomasjpfan committed:
[MRG] ENH Does not overwrite ylabel in partial dependence if present (#15776)

zhaofeng-shu33 committed:
DOC fix typo in plot_tweedie_regression_insurance_claims.py (#18026)

glemaitre committed:
DOC make warning more obvious in example regarding feature importance (#18024)

lucyleeow committed:
DOC Fix return docstring cross_val_predict (#18035)

JinLi711 committed:
DOC Changed Doc to replace "optional" with "default" (#17982)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

dachosen1 committed:
ENH warn in clustering metrics given continuous input (#17911)

ogrisel committed:
DOC add section on debugging ARM64 problems (#18033)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.estimator_checks.py (#18037)

franslarsson committed:
DOC Fix doc of defaults in sklearn.calibration.py (#18034)

lucyleeow committed:
DOC Add doc for permutation_test_score (#17373)

rth committed:
Arm64 CI setup with TravisCI (#17996)

lucyleeow committed:
DOC Add example of using CV iterator to split train test (#17550)

lucyleeow committed:
DOC Update plot_permutation_test_for_classification.py (#17385)

tijanajovanovic committed:
DOC cleaning parameter docstrings in feature_extraction/image.py (#17900)

Co-authored-by: gagileo <babic.dr@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

ckastner committed:
TST: Use assert_allclose instead of equality in FP comparison (#18053)

lucyleeow committed:
DOC Use math notation in log_loss metric (#18049)

franslarsson committed:
DOC update docstring default values in KernelDensity to follow doc guideline (#18050)

cmarmo committed:
FEA Add iterable values support for dictvectorizer (#17367)



Co-authored-by: Peng Yu <peng.yu@shopify.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.validation.py (#18048)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

franslarsson committed:
DOC Fix doc of defaults in sklearn.metrics._scorer.py (#18058)

thomasjpfan committed:
DOC Improves sidebar navigation for user guide (#16964)

justmarkham committed:
DOC Minor documentation fixes (#18056)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.graph.py (#18064)

franslarsson committed:
DOC Fix doc of defaults in manifold._spectral_embedding.py (#18061)

franslarsson committed:
DOC Fix doc of defaults in sklearn.manifold._mds.py (#18060)

glemaitre committed:
Update whats new formatting

ogrisel committed:
MAINT mark AgglomerativeClustering's new compute_distances as |Enhancement|

glemaitre committed:
DOC update the whats new (#18070)

thomasjpfan committed:
DOC Adds whats new to 0.23.2 for bug fix (#17912)

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
TST More cleaning / refactoring in KMeans tests part 2 (#17973)

* remove test_minibatch_set_init_size already included in test_minibatch_init_size

* remove test_minibatch_tol test nothing

* improve and parametrize test_predict_dense_sparse

* remove test_full_vs_elkan already in test_predict and test_elkan_results

* simplify test_n_init

* fix test_centers_not_mutated

* cleaner test_kmeans_init_fitted_centers

* simpler test_kmeans_iter_attribute

* parametrize test_kmeans_empty_cluster_relocated dense sparse

* cln test_warning_elkan_1_cluster

* single double quote uniformisation

* add comment for test_euclidean_distance

* parametrize test_sample_weight_unchanged

* join test_predict, test_fit_predict and test_predict_equal_labels

* remove test_sparse_mb_k_means_callable_init & test_mini_batch_k_means_random_init_partial_fit already included it test_all_init

* Estimator -> algorithm

ogrisel committed:
MAINT scipy-dev failures because of too numerically strict tests (#18071)

solegalli committed:
DOC Add Feature engine to Related Projects page (#18012)

Co-authored-by: Soledad Galli <solegalli1@gmail.com>

glemaitre committed:
MNT update date and links on the index webpage (#18081)

sadakmed committed:
DOC Fix doc of defaults in gaussion_process._gpr.py (#18075)

ckastner committed:
TST Increase absolute tolerance for RANSACRegressor test (#18082)

An absolute difference of 1.45081487e-13 has been observed on 64-bit PowerPC.

franslarsson committed:
DOC Fix doc of defaults in sklearn.metrics._ranking.py (#18093)

FelixWick committed:
FEA allow unknowns in OrdinalEncoder transform (#17406)

jnothman committed:
DOC Some what's new tweaks

NicolasHug committed:
MNT remove pre-dispatch and verbose from CalibratedClassifierCV (#18030)

bmaisonn committed:
DOC Fix documentation of default values in fastica.py (#17989)

bmaisonn committed:
Fix MemoryError exception when calling barycenter_weights with large inputs (#17997)

xun-tang committed:
DOC Address numpydoc validation to GradientBoosting* (#15477)

Co-Authored-By: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Matteo Visconti dOC <mvdoc@berkeley.edu>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

hiteshsom committed:
DOC Update tree.rst about lack of support of categorical data (#18106)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._theil_sen (#18099)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._sag.py (#18098)

NicolasHug committed:
[MRG] Prototype 4 for strict check_estimator mode (#17361)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._omp.py (#18100)

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

thomasjpfan committed:
TST Sets random_state in test_rfe_wrapped_estimator (#18097)

cmarmo committed:
FEA Add Nonnegative LinearRegression (#17578)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>
Co-authored-by: Joseph Knox <jknox13@uw.edu>
Co-authored-by: Joseph Knox <joseph.edward.knox@gmail.com>

wmvanvliet committed:
ENH Add alpha_per_target parameter to RidgeCV (#6624)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

cmarmo committed:
DOC Fix mpl in plot_partial_dependence_visualization_api (#18109)

bharatr21 committed:
ENH: Add regularization to the main NMF class (#17414)

* ENH: Add regularization to the main NMF class

* Update _nmf with suggestions from code review

Update `_nmf.py` with suggestion from review

Co-authored-by: Adrin Jalali <adrin.jalali@gmail.com>

* Refactor tests, fix linter errors

* Change default value to None

* Revert back to default value of "both"

* Update default value documentation acc to @thomasjpfan

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* CLN Places regularization at the end

* Add whatsnew entry

* DOC Fix

* DOC Fix

Co-authored-by: Adrin Jalali <adrin.jalali@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

ckastner committed:
TST Change expected result type np.int64 -> np.int (#18089)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

timokau committed:
TST Relax init parameter checks for unmutable objects (#17936)

KartikChugh committed:
EXA Improves contour boundaries for K-Nearest Neighbors (#17580)

bharatr21 committed:
DOC Correction in empirical covariance (#17587)

thomasjpfan committed:
ENH Adds support for list of bools in column transformer (#17616)

kaylani2 committed:
DOC Mention the conditions for a ValueError in VarianceThreshold  (#17380)

NicolasHug committed:
DOC minor typo fix (#18119)

tbenjamin-pl committed:
DOC Fixes hard coded title in gmm selection example plot (#17418)

Co-authored-by: Thomas S. Benjamin <tbenjamin@vencorelabs.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

hcars committed:
DOC Add note on the max number of extmath.cartesian arguments (#16406)

Co-authored-by: Henry <hlc5v@virginia.edu>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

sstalley committed:
ENH Add verbose option to SpectralClustering (#18052)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

cmarmo committed:
DOC Mark add_toctree_function safe for parallel read and write (#18123)

martina-zxy committed:
DOC Fix doc of default values in sklearn.neural_network._multilayer_perceptron.py (#18133)

* DOC Fix doc of default values in sklearn.neural_network._multilayer_perceptron.py

* Update sklearn/neural_network/_multilayer_perceptron.py

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Update TSNE docstring (#18120)

GaelVaroquaux committed:
DOC on issue triaging process (#17907)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

NicolasHug committed:
DOC Add example comparing PCR and PLS (#17151)

qzhang90 committed:
FIX Fix a cross-platform endian issue in trees (#17644)

* fixed a cross-platform endian issue

* removed a duplicated dtype checking

* Trigger [arm64] CI

* added an entry to doc/whats_new/v0.24.rst

* moved my fix entry to sklearn.tree section

* Update sklearn/tree/_tree.pyx

Added comments

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* code simplified

* fixed a typo in sklearn/tree/_tree.pyx

* Update doc/whats_new/v0.24.rst

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* updated doc/whats_new/v0.24.rst

* Update sklearn/tree/_tree.pyx

Added a space after "if", to be PEP8 compliant.

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* Update sklearn/tree/_tree.pyx

Co-authored-by: Qi Zhang <q.zhang@ibm.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

smarie committed:
FIX use specific threshold to discard eigenvalues with 32 bits fp(#18149)

Co-authored-by: Sylvain MARIE <sylvain.marie@se.com>

cmarmo committed:
DOC Clarify triaging role. Remove usage question template. (#18142)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

simonamaggio committed:
DEP Deprecate n_classes_ in GradientBoostingRegressor (#17702)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lucyleeow committed:
DOC Expand Brier score, fix docstring (#18051)

* fix brier

* fix brier

* add ref

* add ref for lint

* Add link to PDF for calibration book chapter.

* Fix broken repeated ref by using qualified identifier

* Fix indentation

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

lucyleeow committed:
DOC Add user guide for permutation_test_score (#18055)

* wip

* add user guide

* fix link to ex

* suggestion

* suggestions

thomasjpfan committed:
MNT Support for case insensitive filesystems (#18151)

RoeiKa committed:
ENH Permutation importance support sample weight  (#16906)

thomasjpfan committed:
CI Disable 32bit pytest xdist (#18161)

fujiaxiang committed:
API Change default value of as_frame in fetch_openml to 'auto' (#17610)

cmarmo commented:
Hi @BenjaminBossan , are you still interested in working on that? If yes, do you mind fixing conflicts? This will hopefully bring back some attention from reviewers. Thanks!

rauwuckl committed:
DOC clarify the kernel gradient for GaussianProcesses (#18115)

dmohns committed:
[MRG] Add Detection Error Tradeoff (DET) curve classification metrics (#10591)

* Initial add DET curve to classification metrics

* Add DET to exports

* Fix DET-curve doctest errors

- Sample snippet in  model_evaluation documentation was outdated.

* Clarify wording in DET-curve computation

- Align to the wording of ranking module to make it consistent.
- Add correct describtion of input and outputs.
- Update and fix non-existent links

* Beautify DET curve documentation source

- Limit line length to 80 characters.

* Expand DET curve documentation

- Add an example plot to show difference between ROC and DET curves.
- Expand Usage Note section with background information and properties
of DET curves.

* Update DET-curve documentation

- Fix typos and some grammar improvements.
- Use named references to avoid potential conflicts with other sections.
- Remove unneeded references and improved existing ones by using e.g.
using versioned links.

* Select relevant DET points using slice object

* Remove some dubiety from DET curve doc-string

* Add DET curve contributors

* Add tests for DET curves

* Streamline DET test by using parametrization

* Increase verbosity of DET curve error handling

- Explicitly sanity check input before computing a DET curve.
- Add test for perfect scores.
- Adapt indentation style to match the test module.

* Add reference for DET curves in invariance test

* Add automated invariance checks for DET curves

* Resolve merge artifacts

* Make doctest happy

* Fix whitespaces for doctest

* Revert unintended whitespace changes

* Revert unintended white space changes #2

* Fix typos and grammar

* Fix white space in doc

* Streamline test code

* Remove rebase artifacts

* Fix PR link in doc

* Fix test_ranking

* Fix rebase errors

* Fix import

* Bring back newlines

- Swallowed by copy/paste

* Remove uncited ref link

* Remove matplotlib deprecation warning

* Bring back hidden reference

* Add motivation to DET example

* Fix lint

* Add citation

* Use modern matplotlib API

Co-authored-by: Jeremy Karnowski <jeremy.karnowski@gmail.com>
Co-authored-by: Julien Cornebise <julien@cornebise.com>
Co-authored-by: Daniel Mohns <daniel.mohns@zenguard.org>

Commit:
Merge branch 'calibratedclassifiercv-passes-on-fit-params' of https://github.com/BenjaminBossan/scikit-learn into calibratedclassifiercv-passes-on-fit-params

Commit:
Fix linting error, line too long
